src/acvp_kdf108.c

```diff
@@ -221,89 +275,126 @@ ACVP_RESULT acvp_kdf108_kat_handler (ACVP_CTX *ctx, JSON_Object *obj) {
         groupval = json_array_get_value(groups, i);
         groupobj = json_value_get_object(groupval);
 
-        kdf_mode_str = (char *)json_object_get_string(groupobj, "kdfMode");
-        mac_mode_str = (char *)json_object_get_string(groupobj, "macMode");
-        key_out_bit_len = json_object_get_number(groupobj, "keyOutLength");
-        ctr_len = json_object_get_number(groupobj, "counterLength");
-        ctr_loc_str = (char *)json_object_get_string(groupobj, "counterLocation");
-
-        // Get the keyout byte length  (+1 for overflow bits)
-        key_out_len = (key_out_bit_len + 7) / 8;
-
-        if (key_out_len > ACVP_KDF108_KEYOUT_MAX) {
-            ACVP_LOG_ERR("ACVP server requesting unsupported key_out length (%d)",
-                         key_out_len);
-            return (ACVP_UNSUPPORTED_OP);
+        kdf_mode_str = json_object_get_string(groupobj, "kdfMode");
+        if (!kdf_mode_str) {
+            ACVP_LOG_ERR("Failed to include kdfMode");
+            return ACVP_MISSING_ARG;
         }
 
         /*
          * Determine the KDF108 mode to operate.
          * Compare using protocol specified strings.
          */
-        if (strncmp(kdf_mode_str, ACVP_MODE_COUNTER, 64) == 0) {
+        if (strncmp(kdf_mode_str, ACVP_MODE_COUNTER,
+                    strlen(ACVP_MODE_COUNTER)) == 0) {
             kdf_mode = ACVP_KDF108_MODE_COUNTER;
-        } else if (strncmp(kdf_mode_str, ACVP_MODE_FEEDBACK, 64) == 0) {
+        } else if (strncmp(kdf_mode_str, ACVP_MODE_FEEDBACK,
+                           strlen(ACVP_MODE_FEEDBACK)) == 0) {
             kdf_mode = ACVP_KDF108_MODE_FEEDBACK;
-        } else if (strncmp(kdf_mode_str, ACVP_MODE_DPI, 64) == 0) {
+        } else if (strncmp(kdf_mode_str, ACVP_MODE_DPI,
+                           strlen(ACVP_MODE_DPI)) == 0) {
```

Comment:
(strlen --> strnlen) I assume these will be part of this later user story to sweep through all these? I think most of them were converted when safec was integrated, but I guess we will need another pass at some point

Reply:
Yeah I think doing all of them in a dedicated story is best